### PR TITLE
v1.9 backport: node: Remove check whether nextHop is in same L2

### PR DESCRIPTION
**Pull request by brb**
This is a backport of #14354 for v1.9.

```release-note
node: Remove check whether nextHop is in same L2
```

### CODE REVIEW

**Comment by brb** [comment]
test-backport-1.9

**Review by aanm — APPROVED** [review]

File: pkg/datapath/linux/node.go [hunk 1]
```diff
@@ -576,6 +577,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 			// a gateway. Send arping to the gw IP addr instead of newNode IP addr.
 			// NOTE: we currently don't handle multipath, so only one gw can be used.
 			copy(nextHopIPv4, route.Gw)
+			copy(srcIP, route.Src)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]: We don't know if `route.Src` as size 4. We need to either make `srcIP` with size `net.IPv6Len` or do:
```
routeSrcIPv4 := route.Src.To4()
```

File: pkg/datapath/linux/node.go [hunk 2]
```diff
@@ -555,6 +555,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 		return
 	}
 
+	srcIP := make(net.IP, 4)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]:
```
srcIP := make(net.IP, net.IPv4len)
```

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/datapath/linux/node.go [hunk 3]
```diff
@@ -576,6 +577,7 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 			// a gateway. Send arping to the gw IP addr instead of newNode IP addr.
 			// NOTE: we currently don't handle multipath, so only one gw can be used.
 			copy(nextHopIPv4, route.Gw)
```
Comment (aanm) [inline pkg/datapath/linux/node.go]: same here for the `To4()`, no?

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by brb** [comment]
test-backport-1.9

**Review by aanm — APPROVED** [review]

**Comment by brb** [comment]
retest-net-next

**Comment by brb** [comment]
retest-runtime

**Comment by brb** [comment]
CI net-next failed with:
```
/home/jenkins/workspace/Cilium-PR-K8s-1.13-net-next/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Can not connect to service "http://192.168.36.11:31362" from outside cluster
Expected command: kubectl exec -n kube-system log-gatherer-784hz -- curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 20 http://192.168.36.11:31362 -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 1 
Err: exit status 1
Stdout:
 	 
Stderr:
 	 error: unable to upgrade connection: Authorization error (user=kube-apiserver-kubelet-client, verb=create, resource=nodes, subresource=proxy)
```

**Comment by brb** [comment]
retest-net-next

**Review by pchaigno — APPROVED** [review]
The two backported commits are missing references to the upstream commits. Those references are automatically added when using our cherry-pick script.

@brb If you want to fix that, I can merge without rerunning the whole tests after the update. Otherwise, I'll merge before end of day.

**Comment by brb** [comment]
> The two backported commits are missing references to the upstream commits. Those references are automatically added when using our cherry-pick script.

There are no upstream commits. The commits in this PR are only for v1.9.

**Comment by pchaigno** [comment]
> > The two backported commits are missing references to the upstream commits. Those references are automatically added when using our cherry-pick script.
> 
> There are no upstream commits. The commits in this PR are only for v1.9.

Oh, I missed that the "upstream" PR was closed and I had completely forgot our discussion. Merging both backport PRs :ship:

**Pull request merged**